### PR TITLE
Update style of SynExpr.TryWith with a single clause

### LINKED ISSUE #1881: Try/With with a single clause
Issue created from [fantomas-online](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKB6dYDyBpAOgE74B2ALoQJ5GkQQAOihAzgJbNkDKZAhmYmGgBXQoUTlufRDRIB3VmQAWYRAA8wAWgB8MumHqFW5AGYkw%2BSJ1gBbREqMBzMLPFlnhWCQdoApAEELFVUZGUwcAmJyKl06RhZ2Ll5%2BQRExCSTpSPklMBQAfjBOSg5EawA6P3p6ABtWaD5WTwBRVWhEejJGsx5mIM0dSL0DIzJTc0sbO0VHZ1d3T28wf0C1SAAacChmO06vZlQwAG1NiFA9CEgAEjJKRgPIEfWTuiuANx5qoWQ0Y-O6AAYNn8LgBJEhICSsABeyCBwIALM8IABdZ4AXzhpyRUGut2%2BUEemJel3en3xv2BAEYiXpIABZHiqAAyRkQTPEDiUT2BYEpACZ-tjUXoMc8zucrjc7mhIAAjWCwarciUkj5fA4Uv58mkvTilOqKzx%2BMhNcHYYwskiw7EQYwfbZC9FE8W03HSqDyxXK12k9U-G1gADMOounHoPDaACFEMZYGIAAo8Qg8Wz8Qje84UL6OkXO7GSvH3T1KnVvNXkgPwkNQMMRxDR2NiJmwFyEerbMGvWD1TqeDN6LOIHN0UV6F3EqX4uUKkv51VkjUBgCs1cs4ajMbjiAAqlUmO3EJ3uw0%2B9W7dUHX9hSO838C%2B7p17S-O-UcAwA2Ve1jeNxAAYWqHpmD-TwOEIIRoDION%2Bzoc9L3Oa8IFHOhxwuN0p2LGCywXf0eQAdi-dd603MQ6VKWUmBg217SHK8nTFOdJyLGcsJfCseQADkIusGy3EDqlPAM4NohD6LHRjCxlTDn19djgQATm4to-GMNMQOsaweCosBB2HJDbxVJipJYmTy0XHlKUBAM1x4kjED1awDQEkhtOEvSwGQrE73Q5inznWTzKpalrO-RAVLTBynMEnldLo3MGO8oyPRM-yzNwqltWssEITIbASAAFSoAB1BRFFcmj3M8sBUJxJLH1nbyAvSv5KWDEKiL8DwhHBAARRBakctNtNi0T4vExLJIJchWKat8LKrayGVUEFjHyxRxCaC97MUOMyBK%2BAuWreFBTim8EsMyaHmm0ycLmqkV0WxkwWMVhVGwOI%2BDjFoDEQZg2Gi4ElxO0azvGi6H0JVLbs1c5KU-R7VAAJUQOBCHgfbDsrYG9EQjyDNdOrIcatK7pagiEeR1H4AAOSEawKMIc0QX4ax9mrSlKvxidLusIRqk6WorQAMTjTSyDTfLCxu18Yb0SkuOsym43gOk%2BYF1kRcIMWhtXaBFCTCM0wAfXkA6ys586CcuomVVm2W6EpRSEc65NKGwQgWQ4DGyqO7G6FxqqavvKcbZ9En7YgAVVyWl2eDdj2Elp%2BmmCZlm2YDDnTv0y3uYfXn%2BdYQXEE17WmElu5pbkrVgp5SBY-jz2yFVgui5LvgdesvWDcgpgTdYM3IAtsGrYh66oZlgM%2BUy2uloANXLSMjHgRxve046h5QiTR7IGbw8ntqZ8ZIXusgrpF-BFf%2B8xnl16zvGc7Qwmx%2BJ6HJ4Ww-VB62AyAAcTsb6xB-S6KvasQMN5eXBhhFKL8J48j5A9GeatC6skjAJaAABrSMyYMF2GYHlTgKZ-yKjpi5M8FU76By3lAvyMCq7nD5PDWu1NECyCLtGMgLhxDl0QH1F6JAFBdD8OCMiycWDlS2uA6qVDfINVtnvOB5Na64EQO0Faa1xBggIbYS01oeRuQoVzR%2B1tn5yNfnAhWH9NqOWYIoEBWNJFBx8sZGhpjYHAj5E7WunBHDVDCoQBwdNXBFUQLKOkCBdHAn0SDbOw9c7UNkWHMxwJAxWVrn4WoDgSDHxIKfTw3jMl8BEIgfKsBsquBPKQoS5Don31iUYh80lx50L0IGGuwI678yYCQBorw2QLBEQzXhRgBGgXEfBHGYlN4TQadA1xzS6CBmnu05unQdFMhTLKeAPAAKwDYF4ZhrDWRjJEhMsaUzIEyN3kkv4gYD7tJ6uwHgspfFWPYIoIo5BGTHIcdImU4h4Dmh0dwSgvirluJue-dppoAUWiObrQg1RjCDwMQ-Wql1Gm0MCjchB7TlGqPBK4MEWCeA5J9lUiRKK6lopmS4xJ4LziBkYe01BJL0E6OYJ1WA3Uaa-X4CrJBRcAG-X%2BiQdOMVwInP9pMiBI94lgvmRAQMijmVJl4mIB5zBoCGEct0vl25%2BGeD6tAQCyZeyVL0dU05oNzmysuZXLFDKLHtO4IYSCYSkDfIoZsZEcIHjMCFmwA4bk0RAA)

#### Code

```fsharp
// OK
try
    persistState currentState
with ex ->
    printfn "Something went wrong: %A" ex

// OK
try
    persistState currentState
with :? System.ApplicationException as ex ->
    printfn "Something went wrong: %A" ex
```
            

#### Result

```fsharp
// OK
try
    persistState currentState
with
| ex -> printfn "Something went wrong: %A" ex

// OK
try
    persistState currentState
with
| :? System.ApplicationException as ex -> printfn "Something went wrong: %A" ex

```
            
#### Problem description

The [MS F# style guide](https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting#formatting-trywith-expressions) was updated recently and Fantomas does no longer adhere to it.

To solve this issue, a new setting will have to be introduced to still adhere to the [GR style guide](https://github.com/G-Research/fsharp-formatting-conventions#formatting-trywith-expressions).
The guides diverge on this topic.

//cc @Smaug123 @pbiggar 

#### Extra information

- [ ] The formatted result breaks by code.
- [ ] The formatted result gives compiler warnings.
- [ ] I or my company would be willing to help fix this.

#### Options

Fantomas Master at 09/07/2021 15:59:52 - 7d861f4834c9bec59a1f3929265ecc3fc600c196

Default Fantomas configuration

<sub>Did you know that you can ignore files when formatting from fantomas-tool or the FAKE targets by using a [.fantomasignore file](https://github.com/fsprojects/fantomas/blob/master/docs/Documentation.md#ignore-files-fantomasignore)?</sub>

### PR DESCRIPTION
**Pull request by nojaf**
Fixes #1881.
@dsyme could you review my PR comments please?
I have some questions in regards to https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting#formatting-trywith-expressions.

### CODE REVIEW

File: src/Fantomas.Core.Tests/ColMultilineItemTests.fs [hunk 1]
```diff
@@ -135,8 +135,8 @@ let b (x: int) (y: int) : int =
 let c () =
     try
         0
-    with
-    | ex -> 1
+    with ex ->
```
Comment (nojaf) [inline src/Fantomas.Core.Tests/ColMultilineItemTests.fs:138]: Should the body of the `SynMatchClause` always be on the next line, indented?

Comment (dsyme) [reply src/Fantomas.Core.Tests/ColMultilineItemTests.fs:138]: Yes

File: src/Fantomas.Core.Tests/ControlStructureTests.fs [hunk 2]
```diff
@@ -819,7 +818,8 @@ try
     foo.CreationTime <> defaultTime
 with
 // hmm
-| :? FileNotFoundException -> false
+| :? FileNotFoundException ->
```
[the remark this answers is not in the document]
Comment (dsyme) [reply src/Fantomas.Core.Tests/ControlStructureTests.fs:821]: Yes, that's ok

File: src/Fantomas.Core.Tests/MultilineFunctionApplicationsInConditionExpressionsTests.fs [hunk 3]
```diff
@@ -95,11 +95,10 @@ module ElectrumClient =
                     stratumClient.ServerVersion
                         CLIENT_NAME_SENT_TO_STRATUM_SERVER_WHEN_HELLO
                         PROTOCOL_VERSION_SUPPORTED
-                with
-                | :? ElectrumServerReturningErrorException as except when
-                    except.Message.EndsWith
-                        (PROTOCOL_VERSION_SUPPORTED.ToString())
-                    ->
+                with :? ElectrumServerReturningErrorException as except when
```
[the remark this answers is not in the document]
Comment (dsyme) [reply src/Fantomas.Core.Tests/MultilineFunctionApplicationsInConditionExpressionsTests.fs:98]: This feels like it should be multiline formatted - anything with a `when` is too complex for single-line format?

**Review by dsyme — COMMENTED** [review]
Replied to questions on PR

File: src/Fantomas.Core.Tests/MultilineFunctionApplicationsInConditionExpressionsTests.fs (hunk 3 above)
[the remark this answers is not in the document]
Comment (nojaf) [reply src/Fantomas.Core.Tests/MultilineFunctionApplicationsInConditionExpressionsTests.fs:98]: I'm going to circle back to this once we have closure in https://github.com/fsharp/fslang-design/issues/662.

**Pull request merged**